stackslib/src/clarity_vm/clarity.rs

```diff
@@ -1565,7 +1590,7 @@ impl<'a> ClarityBlockConnection<'a, '_> {
         })
     }
 
-    pub fn start_transaction_processing<'c>(&'c mut self) -> ClarityTransactionConnection<'c, 'a> {
+    pub fn start_transaction_processing<'c>(&'c mut self) -> ClarityTransactionConnection<'c, 'c> {
```

Comment:
I need to take a closer look at this. Do we lose anything by making these two lifetimes the same? Also, if this stays like this, I think we can remove all of the lifetimes and let the compiler imply them.

Reply:
```rust
pub struct ClarityBlockConnection<'a, 'b> {
    datastore: WritableMarfStore<'a>,
    header_db: &'b dyn HeadersDB,
    burn_state_db: &'b dyn BurnStateDB,
    cost_track: Option<LimitedCostTracker>,
    mainnet: bool,
    chain_id: u32,
    epoch: StacksEpochId,
}
```

Reply:
I think the first lifetime covers the second one. I've removed the lifetimes.